python/ray/serve/_private/application_state.py

```diff
@@ -390,106 +425,80 @@ def _determine_app_status(self) -> Tuple[ApplicationStatus, str]:
-            try:
-                args, err = ray.get(self._build_app_task_info.obj_ref)
-                if err is None:
-                    logger.info(f"Deploy task for app '{self._name}' ran successfully.")
-                    return (args, config_version), BuildAppStatus.SUCCEEDED, ""
-                else:
-                    error_msg = (
-                        f"Deploying app '{self._name}' failed with "
-                        f"exception:\n{err}"
-                    )
-                    logger.warning(error_msg)
-                    return None, BuildAppStatus.FAILED, error_msg
-            except RuntimeEnvSetupError:
-                error_msg = (
+        # Retrieve build app task result
+        self._build_app_task_info.finished = True
+        try:
+            args, err = ray.get(self._build_app_task_info.obj_ref)
+            if err is None:
+                logger.info(f"Deploy task for app '{self._name}' ran successfully.")
+            else:
+                return (
+                    None,
+                    BuildAppStatus.FAILED,
+                    (f"Deploying app '{self._name}' failed with " f"exception:\n{err}"),
+                )
+        except RuntimeEnvSetupError:
+            return (
+                None,
+                BuildAppStatus.FAILED,
+                (
                     f"Runtime env setup for app '{self._name}' failed:\n"
                     + traceback.format_exc()
-                )
-                logger.warning(error_msg)
-                return None, BuildAppStatus.FAILED, error_msg
-            except Exception:
-                error_msg = (
+                ),
+            )
+        except Exception:
+            return (
+                None,
+                BuildAppStatus.FAILED,
+                (
                     f"Unexpected error occured while deploying application "
                     f"'{self._name}': \n{traceback.format_exc()}"
-                )
-                logger.warning(error_msg)
-                return None, BuildAppStatus.FAILED, error_msg
+                ),
+            )
 
-        return None, BuildAppStatus.IN_PROGRESS, ""
+        # Convert serializable deployment args to deployment infos and
+        # apply option overrides from config
+        try:
+            deployment_infos = {
+                params["deployment_name"]: deploy_args_to_deployment_info(
+                    **params, app_name=self._name
+                )
+                for params in args
+            }
+            overrided_infos = override_deployment_info(
+                self._name, deployment_infos, self._target_state.config
+            )
+            self._route_prefix, self._docs_path = self._check_routes(overrided_infos)
+            return overrided_infos, BuildAppStatus.SUCCEEDED, ""
+        except (TypeError, ValueError, RayServeException):
+            return None, BuildAppStatus.FAILED, traceback.format_exc()
```

Comment:
would suggest to catch any other unexpected exception type too (there may be some validation corner case that raises something else)